src/docusign/util/generation/envelope-preview.ts

```diff
@@ -42,16 +44,44 @@ import { recipientView } from '../../recipient-view-request';
  */
 export const generateEnvelopeAndPreview = async (user, weekId, authArgs, googleParent, redirectUrl) => {
 
-  const week = await getConnection().getRepository(Week).findOne({ where: { id: weekId }});
-  const days = await getConnection().getRepository(Day).find({ where: { user: user, weekId: weekId }, order: { day: 'ASC' }});
-  const supervisor = await getConnection().getRepository(User).findOne({ where: { email: user.supervisorEmail }});
+  const week = await getConnection().getRepository(Week).findOne({ where: { id: weekId } });
+  const days = await getConnection().getRepository(Day).find({
+    where: { user: user, weekId: weekId },
+    order: { day: 'ASC' }
+  });
+  const supervisor = await getConnection().getRepository(User).findOne({ where: { email: user.supervisorEmail } });
 
-  if(days.length === 0){
+  if (days.length === 0) {
     throw new HttpException("No times have been logged for any days this week", HttpStatus.BAD_REQUEST);
   }
 
   const daysResult = await timesTo2DArray7Days(user, days);
 
+  const userSignedDate = new Date();
+
+  let submitterTimeString;
+
+  if (userSignedDate.toLocaleString('en-US', { timeZone: 'America/Los_Angeles' }).includes('AM')) {
```

Comment:
@ValdarRudman I think you can get the function to produce the timezone name, which is better because it will give `PST` or `PDT` depending on whether the Daylight Saving applies to that date or not.
Also you can eliminate the `if` statment by repeating the `replace()`:

```typescript
            .toLocaleString('en-US', {
                timeZone: 'America/Los_Angeles' ,
                timeZoneName: 'short'
            })
            .replace(' AM', 'am')
            .replace(' PM', 'pm');
```